Orange/tests/test_naive_bayes.py

```diff
@@ -53,3 +56,160 @@ def test_allnan_cv(self):
         data = Table('voting')
         results = CrossValidation(data, [self.learner])
         self.assertFalse(any(results.failed))
+
+    def test_prediction_routing(self):
+        data = self.data
+        predict = self.model.predict = Mock(return_value=(data.Y, None))
+
+        self.model(data)
+        predict.assert_called()
+        predict.reset_mock()
+
+        self.model(data.X)
+        predict.assert_called()
+        predict.reset_mock()
+
+        self.model.predict_storage(data)
+        predict.assert_called()
+        predict.reset_mock()
+
+        self.model.predict_storage(data[0])
+        predict.assert_not_called()
+
+    def test_compare_results_of_storage_and_predict_storage(self):
+        data2 = NotATable("titanic")
+
+        self.model = self.learner(self.data[:50])
+        values, probs = self.model.predict_storage(self.data[50:])
+        values2, probs2 = self.model.predict_storage(data2[50:])
+        np.testing.assert_equal(values, values2)
+        np.testing.assert_equal(probs, probs2)
+
+    def test_predictions(self):
+        self._test_predictions(sparse=None)
+
+    def test_predictions_csr_matrix(self):
+        with warnings.catch_warnings():
```

Comment:
I preferred silencing this warnings (which will be removed once `scipy.sparse` stops using the matrix subclass or silences the warnings there) here rather than in `_test_predictions` to save horizontal space there.